theano/gof/graph.py

```diff
@@ -118,7 +117,8 @@ def __init__(self, op, inputs, outputs):
                 if output.owner is None:
                     output.owner = self
                     output.index = i
-                    output.tag = utils.scratchpad()
+                    if not hasattr(output, "tag"):
+                        output.tag = utils.scratchpad()
```

Comment:
Why not remove those 2 lines? Why they are needed?

I'm ok with the change, but I don't understand why it was like this first.